### ISSUE TITLE
A minimal implementation of SimCLR on an ImageNet subset

### ISSUE THREAD

**Issue by sayakpaul** [google-research/simclr#21]
I am Sayak, an ML Engineer from India. I am writing to let you know that I have started working on a minimal implementation of SimCLR. 

I was able to use the utility functions (the augmentation policies, and the NT-XEnt loss) provided in the GitHub repository mentioned in the paper. For my implementation, I am using a combination of tf.keras and custom loops with tf.GradientTape. I am trying my experiments on a small subset of ImageNet, **top 5 categories**, **each class having 250 images**. I am unsure at this point if my implementation is buggy but I am not able to get desired results when I am evaluating the self-supervised learned representations using Linear Evaluation. 

This is why I am reaching out to know if it'd be possible to take a look at it and let me know about the feedback. I know this is a lot to ask for. But in case, if you would like to have a look, I have attached two notebooks: 
- one that gathers the dataset, preprocesses it, and trains the SimCLR model along with the major things
- the other one evaluates the learned representations

**I used a GCP VM pre-configured with TensorFlow 2.1 and Tesla T4 GPU.**

[aminimalimplementationofsimclronanimagenetsubset.zip](https://github.com/google-research/simclr/files/4535081/aminimalimplementationofsimclronanimagenetsubset.zip)

**Comment by jerofad** [comment]
Hi, Sayak, I suggest you put it on GitHub and share the link so it is easy for people to look through it and give comment.

**Comment by sayakpaul** [comment]
Here you go: https://github.com/sayakpaul/SimCLR-in-TensorFlow-2.
